doc/PostgreSQL.txt

```diff
@@ -0,0 +1,129 @@
+PostgreSQL Backend
+==================
+
+The PostgreSQL backend is not for everyone. It requires some experience with
+PostgreSQL or a willness to learn. This is not something for the casual user.
+
+Status
+------
+This is an initial release, with improvements, and enhancements to come. Right
+now there is little more than the blocks table, but performance has been good.
+In time there will be mods that will make more use of the database, rather than
+flat files.
+
+Todo
+----
+- Adapt rollback to work with PostgreSQL, since rollback normally used SQLite3
+- Adapt mapgen to use PostgreSQL
+- Test under Windows
+- Adapt mods to use PostgreSQL rather than flat files, when possible
+
+How to help
+-----------
+Please email if this document could be improved in any way or any possible ideas
+that could help.
+
+Who might benefit
+-----------------
+Anyone who is running a dedicated long term server with a regular number of
+players on at all times. Typically this will be a Linux server.
+
+Advantages of PostgreSQL backend:
+---------------------------------
+- Greater use of memory to improve performance
+- Automatic reindexing using the auto vacuum feature
+- Backups while minetest users are playing
+- Continous backups, using automatic transaction log archiving
+- Ability use seperate application and database server, for increased performance
+- Redundancy, load balancing, and failover, via PostgreSQL features
+- Use of triggers and additional conliums or tables
+
+Further documentation
+----------------------
+- PostgreSQL: http://www.postgresql.org/
+- Continous Archiving: http://www.postgresql.org/docs/9.4/static/continuous-archiving.html
+- Tuning: https://wiki.postgresql.org/wiki/Tuning_Your_PostgreSQL_Server
+- Kernel Resources: http://www.postgresql.org/docs/9.4/static/kernel-resources.html
+- Auto Vacuum: http://www.postgresql.org/docs/9.4/static/runtime-config-autovacuum.html
+- High Availability, Load Balancing, and Replication: http://www.postgresql.org/docs/9.4/static/high-availability.html
+
+How to compile
+--------------
+
+This document is not going to be extensive enough to explain the details of
+compiling open source projects. There should be plenty of information on the
+web that explains that much better.
+
+I will explain how to compile Minetest with the PostgreSQL backend on Ubuntu
+14.10. These settings are minimal and meant to be combined with your normal
+biuld configuration.
+
+# apt-get install cmake cmake-data libpq-dev postgresql-client
+
+# cmake -DBUILD_SERVER=TRUE -DENABLE_POSTGRESQL=TRUE -DPostgreSQL_TYPE_INCLUDE_DIR=/usr/include/postgresql/9.3/server/catalog
```

Comment:
That is an issue with cmake's support for PostgreSQL. I have read that it will be fixed. I will submit a patch to them by the end of the week, if it has not been fix at that point.